process/block/preprocess/basePreProcess.go

```diff
@@ -307,3 +312,47 @@ func (bpp *basePreProcess) computeGasConsumedByTx(
 
 	return txGasLimitInSenderShard, txGasLimitInReceiverShard, nil
 }
+
+func (bpp *basePreProcess) saveAccountBalanceForAddress(address []byte) {
+	if bpp.balanceComputation.HasAddressBalanceSet(address) {
+		return
+	}
+
+	balance, err := bpp.getBalanceForAddress(address)
+	if err != nil {
+		//TODO: Analyze if it is better to call return here
+		balance = big.NewInt(0)
+	}
+
+	bpp.balanceComputation.SetBalanceToAddress(address, balance)
+}
+
+func (bpp *basePreProcess) getBalanceForAddress(address []byte) (*big.Int, error) {
+	addressContainer, err := bpp.pubkeyConverter.CreateAddressFromBytes(address)
+	if err != nil {
+		return nil, err
+	}
+
+	accountHandler, err := bpp.accounts.GetExistingAccount(addressContainer)
+	if err != nil {
+		return nil, err
+	}
+
+	account, ok := accountHandler.(state.UserAccountHandler)
+	if !ok {
+		return nil, process.ErrWrongTypeAssertion
+	}
+
+	return account.GetBalance(), nil
+}
+
+func (bpp *basePreProcess) getTxTotalCost(txHandler data.TransactionHandler) *big.Int {
```

Comment:
This is a bit misleading, I thought it will invoke the txFeeHandler. Might rename this to getMaxTxTotalCost

Reply:
done